packages/@aws-cdk/aws-ec2/lib/nat.ts

```diff
@@ -121,16 +144,25 @@ class NatGateway extends NatProvider {
 
     // Add routes to them in the private subnets
     for (const sub of options.privateSubnets) {
+      const az = sub.availabilityZone;
+      const gatewayId = gatewayIds.pick(az);
       sub.addRoute('DefaultRoute', {
         routerType: RouterType.NAT_GATEWAY,
-        routerId: gatewayIds.pick(sub.availabilityZone),
+        routerId: gatewayId,
         enablesInternetConnectivity: true,
       });
+      this.gateways.push({ az, gatewayId });
```

Comment:
This is interesting.

You have decided to add the gateways not as they are CREATED (maybe fewer gateways than AZs available), but as they are SELECTED (there will always be one record per AZ, and it may be the same gateway multiple times).

I would have chosen the opposite (just return the gateways that were created) so downstream code can decide what to do with that and it has all available information.

Why did you choose this solution?

Reply:
Just follow-up a previous VPC api that return a gateway per az. I thought selected GW fits better because they are in use already. For the general design, I do agree that all created GW shall be visible.